nipype/interfaces/fsl/utils.py

```diff
@@ -26,11 +26,56 @@
 from ..base import (traits, TraitedSpec, OutputMultiPath, File,
                     CommandLine, CommandLineInputSpec, isdefined)
 from ...utils.filemanip import (load_json, save_json, split_filename,
-                                fname_presuffix)
+                                fname_presuffix, copyfile)
 
 warn = warnings.warn
 warnings.filterwarnings('always', category=UserWarning)
 
+class CopyGeomInputSpec(FSLCommandInputSpec):
+    in_file = File(exists=True, mandatory=True, argstr="%s", position=0,
+                   desc="source image")
+    dest_file = File(exists=True, mandatory=True, argstr="%s", position=1,
+                     desc="destination image", copyfile=True)
+    ignore_dims = traits.Bool(desc=('Do not copy image dimensions'),
+                              argstr='-d', position="-1")
+
+class CopyGeomOutputSpec(TraitedSpec):
+    out_file = File(exists=True, desc="image with new geometry header")
+
+class CopyGeom(FSLCommand):
+    """Use fslcpgeom to copy the header geometry information to another image.
+
+    Copy certain parts of the header information (image dimensions, voxel dimensions,
+    voxel dimensions units string, image orientation/origin or qform/sform info)
+    from one image to another. Note that only copies from Analyze to Analyze
+    or Nifti to Nifti will work properly. Copying from different files will result
+    in loss of information or potentially incorrect settings.
+
+    """
+    _cmd = "fslcpgeom"
+    input_spec = CopyGeomInputSpec
+    output_spec = CopyGeomOutputSpec
+
+    def _run_interface(self, runtime):
```

Comment:
I see. So I don't know why it is not working either since the command you wrote return the correct path to the file. Here is the full code and output.
```
from nipype.interfaces.fsl.utils import CopyGeom
cpg = CopyGeom(in_file='../test_fslcpgeom_satra/SUBJ01_FAfsl.nii.gz',
               dest_file='../test_fslcpgeom_satra/SUBJ01_T1_brainseed.nii')
cpg_res = cpg.run()
```
```
> cpg_res.outputs.out_file
<undefined>
```
```
> cpg._filename_from_source('dest_file')
 '../test_fslcpgeom_satra/SUBJ01_T1_brainseed.nii'
```
Does the following have to do anything with it?
```
metadata = dict(name_source=lambda t: t is not None)
traits = cpg.inputs.traits(**metadata)
```
```
> traits
{}
```